File: apps/web/src/code/python/basics/download_file.py
from e2b import Sandbox

sandbox = Sandbox(template="base")

file_in_bytes = sandbox.download_file("path/to/remote/file/inside/sandbox")  # $HighlightLine
# Save file to local filesystem
with open("path/to/local/file", "wb") as f:  # $HighlightLine
    f.write(file_in_bytes)  # $HighlightLine

sandbox.close()

File: apps/web/src/code/js/basics/download_file.js
import { Sandbox } from 'e2b'
import fs from 'node:fs'

const sandbox = await Sandbox.create({ template: 'base' })

const buffer = await sandbox.downloadFile('path/to/remote/file/inside/sandbox', 'buffer') // $HighlightLine
// Save file to local filesystem
fs.writeFileSync('path/to/local/file', buffer)

await sandbox.close()
